src/ImageService.Business/Commands/ImageNews/CreateImageNewsCommand.cs

```diff
@@ -0,0 +1,107 @@
+using LT.DigitalOffice.Kernel.Constants;
+using LT.DigitalOffice.Kernel.Enums;
+using LT.DigitalOffice.Kernel.FluentValidationExtensions;
+using LT.DigitalOffice.Kernel.Responses;
+using Microsoft.AspNetCore.Http;
+
+namespace LT.DigitalOffice.ImageService.Business.Commands.ImageNews
+{
+  public class CreateImageNewsCommand : ICreateImageNewsCommand
+  {
+    private readonly IAccessValidator _accessValidator;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly IImageNewsRepository _repository;
+    private readonly IDbImageNewsMapper _mapper;
+    private readonly IResizeImageHelper _resizeHelper;
+    private readonly ICreateImageRequestValidator _validator;
+
+    public CreateImageNewsCommand(
+      IAccessValidator accessValidator,
+      IHttpContextAccessor httpContextAccessor,
+      IImageNewsRepository repository,
+      IDbImageNewsMapper mapper,
+      IResizeImageHelper resizeHelper,
+      ICreateImageRequestValidator validator)
+    {
+      _accessValidator = accessValidator;
+      _httpContextAccessor = httpContextAccessor;
+      _repository = repository;
+      _mapper = mapper;
+      _resizeHelper = resizeHelper;
+      _validator = validator;
+    }
+
+    public OperationResultResponse<List<Guid>> Execute(CreateImageRequest request)
+    {
+      if (!(_accessValidator.IsAdmin() ||
+        _accessValidator.HasRights(Rights.AddEditRemoveNews)))
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+
+        return new OperationResultResponse<List<Guid>>
+        {
+          Status = OperationResultStatusType.Failed,
+          Errors = new() { "Not enough rights." }
+        };
+      }
+
+      if (!_validator.ValidateCustom(request, out List<string> errors))
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+
+        return new OperationResultResponse<List<Guid>>
+        {
+          Status = OperationResultStatusType.Failed,
+          Errors = errors
+        };
+      }
+
+      OperationResultResponse<List<Guid>> response = new();
+
+      List<DbImageNews> dbImagesNews = new() { _mapper.Map(request) };
+
+      if (request.EnablePrewiew)
+      {
+        string resizedContent = _resizeHelper.Resize(request.Content, request.Extension);
+
+        if (resizedContent != null)
+        {
+          dbImagesNews.Add(_mapper.Map(request, dbImagesNews[0].Id, resizedContent));
+        }
+        else
+        {
+          _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+
+          response.Status = OperationResultStatusType.Failed;
+          response.Errors = new() { "Image resize failed." };
+          return response;
+        }
+      }
+
+      response.Body = _repository.Create(dbImagesNews);
+      if (response.Body == null)
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
```

Comment:
it is not bad request. Repo cannot save image, it is i don't know internal server or smth like this. I don't know what code you should use here, but not bad request

Reply:
I think this would be a bad request.

Reply:
ok now it is 400, but you can make 1 return here Instead of 2

Reply:
done